config/routes.rb

```diff
@@ -36,9 +36,12 @@
     resources :notes, only: :create
     resources :feedbacks, only: %i[create index]
     resources :addresses, only: %i[update destroy]
-    post :approve
-    post :reject
-    post :certify
+    collection do
+      post :approve
+      post :reject
+      post :certify
+      post :html_to_pdf
+    end
```

Comment:
Could you explain these changes here? The [Rails documentation](https://guides.rubyonrails.org/routing.html#adding-collection-routes) is kind of circular and says "To add a route to the collection, use a collection block", but it doesn't really define what a "collection" is. I think it means it defines the URL under `/services/approve` instead of `/services/:id/approve`, but I'm not 100% confident.

In either case, did you mean to change the routing configuration for the `approve`, `reject`, and `certify` routes? And could this `collection` block be merged with the `collection` block below it?

Reply:
Oops, I misunderstood the collection block's purpose. I have just added the new endpoint to the existing block and moved the other routes out of the collection block.